ZOOKEEPER-4985 Fix zkCleanup.sh script to not fail on non-standard config directories

lfrancke opened the pull request:
https://issues.apache.org/jira/browse/ZOOKEEPER-4985

The zkCleanup.sh script has several issues that make it difficult to use in containerized or non-standard deployments:

The script hardcodes the path to zoo.cfg and fails with confusing error messages when the config file is not in the default location. When the config file is missing, it passes empty strings to PurgeTxnLog, causing confusing error messages

Steps to Reproduce:

* Run ZooKeeper with config in a non-standard location (e.g., /custom/path/zoo.cfg)
* Try to run cleanup:

```
grep: /path/to/zookeeper/bin/../conf/zoo.cfg: No such file or directory
grep: /path/to/zookeeper/bin/../conf/zoo.cfg: No such file or directory
Path '/path/to/zookeeper/bin' does not exist.
Usage:
PurgeTxnLog dataLogDir [snapDir] -n count 
```

lfrancke committed:
ZOOKEEPER-4985 Fix zkCleanup.sh script to not fail on non-standard config directories

lfrancke committed:
Fix shfmt error

ctubbsii commented on bin/zkCleanup.sh:
```diff
@@ -31,27 +31,44 @@
 ZOOBIN="${BASH_SOURCE-$0}"
 ZOOBIN="$(dirname "$ZOOBIN")"
-ZOOBINDIR="$(cd "$ZOOBIN" && pwd)"
+ZOOBINDIR="$(cd "$ZOOBIN" || exit && pwd)"
```
Will this exist the entire script, or just the subshell that the `cd` command is running in?
It might be better to do something like:


```suggestion
ZOOBINDIR="$(cd "$ZOOBIN" && pwd)" || exit
```

This is a little more clear what is going to happen, especially when it comes to the order of operations between `||` and `&&`, which are not immediately obvious to non-bash experts.

ctubbsii commented on bin/zkCleanup.sh:
```diff
@@ -31,27 +31,44 @@
+ZOOBINDIR="$(cd "$ZOOBIN" || exit && pwd)"
 
 if [[ -e "$ZOOBIN/../libexec/zkEnv.sh" ]]; then
-  # shellcheck source=bin/zkEnv.sh
```
Why remove the shellcheck hints? Those help shellcheck find the scripts in the source code when doing validation.

lfrancke committed:
Update bin/zkCleanup.sh

Co-authored-by: Christopher Tubbs <ctubbsii@apache.org>

lfrancke committed:
Address review comments

ctubbsii commented on bin/zkCleanup.sh:
```diff
@@ -31,27 +31,43 @@
 ZOO_LOG_FILE=zookeeper-$USER-cleanup-$HOSTNAME.log
 
 # shellcheck disable=SC2206
 flags=($JVMFLAGS)
```
It looks like `"-Dzookeeper.log.dir=$ZOO_LOG_DIR" "-Dzookeeper.log.file=$ZOO_LOG_FILE"` is added in all cases. So, it could just be added to `flags` here and removed from the subsequent calls:


```suggestion
flags=("-Dzookeeper.log.dir=$ZOO_LOG_DIR" "-Dzookeeper.log.file=$ZOO_LOG_FILE" $JVMFLAGS)
```

lfrancke committed:
Address review comments

- Make the directory handling much smarter and simpler
- Move the common flags into the "flags" variable instead of repeating them over and over

lfrancke committed:
Handle greedy sed matching and whitespace trimming

lfrancke committed:
Merge branch 'master' into feat/ZOOKEEPER-4985-rebased

lfrancke commented:
@ctubbsii ping to see if you're happy to merge this?
